service/assessment/assessment.go

```diff
@@ -423,19 +385,13 @@ func (svc *Service) RegisterAssessmentResultHook(assessmentResultsHook func(resu
 }
 
 // initEvidenceStoreStream initializes the stream to the Evidence Store
-func (svc *Service) initEvidenceStoreStream(URL string, additionalOpts ...grpc.DialOption) (stream evidence.EvidenceStore_StoreEvidencesClient, err error) {
-	log.Infof("Trying to establish a connection to evidence store service @ %v", svc.evidenceStoreAddress.target)
+func (svc *Service) initEvidenceStoreStream(target string, _ ...grpc.DialOption) (stream evidence.EvidenceStore_StoreEvidencesClient, err error) {
+	log.Infof("Trying to establish a stream to evidence store service @ %v", target)
 
-	// Establish connection to evidence store gRPC service
-	conn, err := grpc.Dial(URL,
-		api.DefaultGrpcDialOptions(URL, svc, additionalOpts...)...,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("could not connect to evidence store service: %w", err)
-	}
+	// Make sure, that we re-connect
+	svc.evidenceStore.ForceReconnect()
```

Comment:
I am now not 100 % sure if this really needed (anymore). Yes the initXStream methods are called automatically by GetStream() if no connection is available and the NewStream method inside RPCConnection checks if the connection is available, so maybe this is not needed.

Thinking about it, I think the reasons was that because if the stream receives an EOF, I was not sure if the RPCConnection struct also receives this, so it can set the clientConn to nil. Otherwise the stream EOF'd but the clientConn somehow still exists in an invalid state. Unfortunately, its quite hard to actually test this properly :(